### PR TITLE
[GNA] Fixes for GNA 3.0 library

### PR DESCRIPTION
**Pull request by kbruniec**
### Details:
 - Fix tests which fail after GNA lib replacement (i.e., to 3.0.0.1363)
 - Fix Assertion in split_convolution_with_large_buffer_size for CNN2D ADL tests

### Tickets:
 - 63011

### CODE REVIEW

File: inference-engine/src/gna_plugin/backend/am_intel_dnn.cpp [hunk 1]
```diff
@@ -248,6 +249,16 @@ void GNAPluginNS::backend::AMIntelDNN::InitConvolutional2DComponentPrivate(intel
     ptr_inputs = &comp.ptr_inputs;
     ptr_outputs = &comp.ptr_outputs;
 }
+
+bool GNAPluginNS::backend::AMIntelDNN::isOperationCnnLegacySpecific(const Gna2Operation& op) {
+    // GNA compile target GNA_TARGET_3_0 does not support pooling window < pooling stride
+    return op.Type == Gna2OperationTypeConvolution &&
+        op.NumberOfParameters > (std::max)(PoolStrideParamIdx, PoolWinParamIdx) &&
```
Comment (elilobanova) [inline inference-engine/src/gna_plugin/backend/am_intel_dnn.cpp:256]: why do you use parentheses around std::max here?

File: inference-engine/src/gna_plugin/transformations/split_convolution_with_large_buffer_size.cpp [hunk 2]
```diff
@@ -28,9 +29,22 @@ static bool Convert(std::shared_ptr<ngraph::Node> conv,
     if (input_size <= GNALimitations::bufferMaxSize) {
         return false;
     }
-
-    uint32_t width = conv->get_input_shape(0).back();
-    uint32_t in_channels = conv->get_input_shape(0).at(1);
+    auto& input = conv->get_input_shape(0);
```
Comment (elilobanova) [inline inference-engine/src/gna_plugin/transformations/split_convolution_with_large_buffer_size.cpp:32]: it's better to move this check into the matcher for this transformation, as a predicate in the second argument of convolution wrap_type:
auto conv = ngraph::pattern::wrap_type<ngraph::opset7::Convolution>({ngraph::pattern::any_input(),
        ngraph::pattern::any_input()}, [](const ngraph::Output<ngraph::Node>& node) {
 { put convolution check here }
});

File: inference-engine/src/gna_plugin/backend/am_intel_dnn.cpp (hunk 1 above)
Comment (kbruniec) [reply inference-engine/src/gna_plugin/backend/am_intel_dnn.cpp:256]: I did out of (bad) habit (see https://stackoverflow.com/questions/5004858/why-is-stdmin-failing-when-windows-h-is-included).
I will remove.

File: inference-engine/src/gna_plugin/transformations/split_convolution_with_large_buffer_size.cpp (hunk 2 above)
Comment (kbruniec) [reply inference-engine/src/gna_plugin/transformations/split_convolution_with_large_buffer_size.cpp:32]: OK

**Review by elilobanova — APPROVED** [review]

**Review by nkogteva — APPROVED** [review]

**Pull request merged**